### PR TITLE
Added indicator and option to hide played audiobooks

### LINKED ISSUE #47: Manually mark as "Played/Watched" does nothing
When I open an audiobook that I've already listened to and click on the eye icon at the top right and mark the book as watched/played. The audiobook does not get the blue bar underneath it indicating that it has been listened to.

### LINKED ISSUE #48: [Feature Request] Sort Option to Hide "Played" Audiobooks
It would be cool if you added an option to "hide" audiobooks from showing up that have been marked as "Played"

### PR DESCRIPTION
**Pull request by lks-nbg**
This PR adds a dog-ear for unplayed audiobooks and a filter option in the library menu

There are now three "states" of an audiobook:
- unplayed (viewCount = 0 & progress = 0) → marked with a dog-ear
- currently playing (progress > 0) → marked with progress bar
- played (viewCount > 0) → no indicator

Resolves #47, resolves #48 

I think the behavior of the toggle_watched button should be changed too since it currently does not allow marking an audiobook as unwatched.

### CODE REVIEW

**Comment by lks-nbg** [comment]
I changed the toggle_watched button behavior but I am not sure if the observer is right. 
The button basically works and updates the viewCount in the database and and the server.

File: app/src/main/java/io/github/mattpvaughn/chronicle/features/bookdetails/AudiobookDetailsFragment.kt [hunk 1]
```diff
@@ -163,6 +163,14 @@ class AudiobookDetailsFragment : Fragment() {
                 if (isSyncing) syncIcon.start() else syncIcon.stop()
             }
         }
+
+        // todo: not sure if there is a better way to only observe viewCount and not the whole audiobook
```
Comment (mattttvaughn) [inline app/src/main/java/io/github/mattpvaughn/chronicle/features/bookdetails/AudiobookDetailsFragment.kt]: Yea ideally we want the fragment to have as little logic in it as possible. Something like this in the viewModel would do it:

```
    val isWatched : LiveData<Boolean> = audiobook.map {
        it?.viewCount != 0L
    }

```

or even better

```
    val isWatchedIcon : LiveData<Int> = audiobook.map {
        // double-check the booleans, could be flipped
        if (it?.viewCount != 0L) R.drawable.ic_visibility_off else R.drawable.ic_visibility
    }

```

File: app/src/main/java/io/github/mattpvaughn/chronicle/features/bookdetails/AudiobookDetailsViewModel.kt [hunk 2]
```diff
@@ -527,6 +542,24 @@ class AudiobookDetailsViewModel(
...
+            Toast.LENGTH_LONG)
+        toast.setGravity(Gravity.BOTTOM,0,200)
+        toast.show()
+    }
+
+    private fun setAudiobookUnwatched() {
+        Timber.i("Marking audiobook as unwatched")
+        viewModelScope.launch {
+            bookRepository.setUnwatched(inputAudiobook.id)
+            // anything else to update here?
```
Comment (mattttvaughn) [inline app/src/main/java/io/github/mattpvaughn/chronicle/features/bookdetails/AudiobookDetailsViewModel.kt]: LGTM

File: app/src/main/res/layout/grid_item_audiobook.xml [hunk 3]
```diff
@@ -52,6 +52,17 @@
             app:layout_constraintRight_toRightOf="parent"
             app:layout_constraintTop_toTopOf="parent">
 
+            <ImageView
+                android:id="@+id/not_played_dog_ear"
+                android:layout_width="32dp"
+                android:layout_height="32dp"
+                android:layout_gravity="end|top"
+                android:contentDescription="@string/not_played_dog_ear"
```
Comment (mattttvaughn) [inline app/src/main/res/layout/grid_item_audiobook.xml:60]: 💯

**Review by mattttvaughn — APPROVED** [review]
looks great!

**Comment by mattttvaughn** [comment]
Feel free to update that bit in `AudiobookDetailsFragment` if you want, but not a big sweat. won't cause any performance issues or memory leaks or anything, just a minor code quality thing bordering on personal preference.

If you haven't put up any changes by a day or so I'll go ahead merge and, but if you want to change it after that feel free to just put up another PR

**Comment by lks-nbg** [comment]
Commit text is false. I added isWatchedIcon in the viewModel and renamed the icon since it is not a white square ;-)

**Pull request merged**